source/_integrations/imap.markdown

```diff
@@ -173,19 +175,104 @@ template:
+| Data attribute | Type | Optional | Description |
+| -- | -- | -- | -- |
+| `entry_id` | string | no | The IMAP config entry ID. |
+| `uid` | string |  no | The `uid` of the message to be marked as "seen". To be found in the message's event data. |
+
+#### Action `move` - Move the IMAP message
+
+| Data attribute | Type | Optional | Description |
+| -- | -- | -- | -- |
+| `entry_id` | string | no | The IMAP config entry ID. |
+| `uid` | string |  no | The `uid` of the message to be marked as seen. To be found in the message's event data. |
+| `target_folder` | string | no | The name of the target folder, for example `INBOX.Trash` where the message should be moved to. |
+| `seen` | boolean | yes | If set to `true` this will mark the message as "seen". |
+
+#### Action `delete` - Delete the IMAP message
+
+| Data attribute | Type | Optional | Description |
+| -- | -- | -- | -- |
+| `entry_id` | string | no | The IMAP config entry ID. |
+| `uid` | string | no | The `uid` of the message to be deleted. To be found in the message's event data. |
 
 {% caution %}
-When these actions are used in an automation, make sure the right triggers and filtering are set up. When messages are deleted, they cannot be recovered. When multiple IMAP entries are set up, make sure the messages are filtered by the `entry_id` as well to ensure the correct messages are processed. Do not use these actions unless you know what you are doing.
+When these actions are used in an automation, make sure the right triggers and filtering are set up. When messages are deleted or modified, they cannot be recovered. When multiple IMAP entries are set up, make sure the messages are filtered by the `entry_id` as well to ensure the correct messages are processed. Do not use these actions unless you know what you are doing.
 {% endcaution %}
 
+#### Action `fetch` - Fetch the an IMAP message
+
+Fetches the text body and retrieves metadata about the parts inside the IMAP message.
+
+| Data attribute | Type | Optional | Description |
+| -- | -- | -- | -- |
+| `entry_id` | string | no | The IMAP config entry ID. |
+| `uid` | string |  no | The `uid` of the message to be marked as seen. To be found in the message's event data. |
+
+##### Return values for the `fetch` action
+
+{% configuration_basic %}
+text:
+  description: The plain text version of the fetched email.
+subject:
+  description: The subject of the fetched email.
+sender:
+  description: The sender's email address of the fetched email.
+uid:
+  description: The UID of the message.
+parts:
+  description: Contains a dictionary with metadata about the available parts in the message in case of a multipart message. This allows to fetch and process the complete message text message part, not limited by size. For each available part `content_type` and `content_transfer_encoding` will be provided. In case a file name is set, it will be supplied through the `filename` key. 
```

Comment:
Don't explain this in plain text, instead do like this:
```
parts:
   description: A dictionary with metadata about the available parts in the message in case of a multipart message. This allows to fetch and process the complete message text message part, not limited by size
   type: map
   keys:
      <part>:
         description: Explain the key
         type: map
            keys:
               content_type:
                  description: blabla
               content_transfer_encoding:
                  description:  bleble
               filename:
                  description:  blublu
```